### PR TITLE
[Merged by Bors] - Update dependencies (rustc nightly-2022-04-15, viper v.22.04-release)

### PR DESCRIPTION
**Pull request by viper-admin**
* [ ] Update Viper version to `v.22.04-release`.
* [x] Update rustc version to `nightly-2022-04-15`.
* [x] Manualy update outdated dependencies (see the list below).
* [x] Manualy run `cargo update`.

<details><summary>List of direct outdated dependencies:</summary>

```
$ cargo outdated --root-deps-only --workspace

info: syncing channel updates for 'nightly-2022-04-15-x86_64-unknown-linux-gnu'
info: latest update on 2022-04-15, rust version 1.62.0-nightly (e7575f967 2022-04-14)
info: downloading component 'cargo'
info: downloading component 'llvm-tools-preview'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustc-dev'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'llvm-tools-preview'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustc-dev'
info: installing component 'rustfmt'
    Updating git repository `https://github.com/rust-lang/cargo.git`
analysis
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
log   0.4.14   ---     0.4.16  Normal  ---
syn   1.0.88   ---     1.0.91  Normal  ---

prusti
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
log   0.4.14   ---     0.4.16  Normal  ---

prusti-common
================
Name    Project  Compat  Latest  Kind    Platform
----    -------  ------  ------  ----    --------
config  0.12.0   ---     0.13.1  Normal  ---
log     0.4.14   ---     0.4.16  Normal  ---

viper
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
log   0.4.14   ---     0.4.16  Normal  ---

viper-sys
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
log   0.4.14   ---     0.4.16  Normal  ---

jni-gen
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
log   0.4.14   ---     0.4.16  Normal  ---

vir
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
log          0.4.14   ---     0.4.16  Normal  ---
proc-macro2  1.0.36   ---     1.0.37  Normal  ---
quote        1.0.15   ---     1.0.18  Normal  ---
syn          1.0.88   ---     1.0.91  Normal  ---

vir-gen
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.36   ---     1.0.37  Normal  ---
quote        1.0.15   ---     1.0.18  Normal  ---
syn          1.0.88   ---     1.0.91  Normal  ---

prusti-contracts
================
Name      Project  Compat  Latest  Kind         Platform
----      -------  ------  ------  ----         --------
trybuild  1.0.56   ---     1.0.59  Development  ---

prusti-contracts-internal
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.36   ---     1.0.37  Normal  ---

prusti-specs
================
Name         Project  Compat  Latest  Kind    Platform
----         -------  ------  ------  ----    --------
proc-macro2  1.0.36   ---     1.0.37  Normal  ---
quote        1.0.15   ---     1.0.18  Normal  ---
syn          1.0.88   ---     1.0.91  Normal  ---

prusti-interface
================
Name    Project  Compat  Latest  Kind    Platform
----    -------  ------  ------  ----    --------
config  0.12.0   ---     0.13.1  Normal  ---
log     0.4.14   ---     0.4.16  Normal  ---

prusti-viper
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
log   0.4.14   ---     0.4.16  Normal  ---

prusti-server
================
Name     Project  Compat  Latest   Kind    Platform
----     -------  ------  ------   ----    --------
clap     2.34.0   ---     3.1.8    Normal  ---
log      0.4.14   ---     0.4.16   Normal  ---
reqwest  0.10.10  ---     0.11.10  Normal  ---
tokio    0.2.25   ---     1.17.0   Normal  ---
warp     0.2.5    ---     0.3.2    Normal  ---

prusti-launch
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
toml  0.5.8    ---     0.5.9   Normal  ---

test-crates
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
clap  2.34.0   ---     3.1.8   Normal  ---
log   0.4.14   ---     0.4.16  Normal  ---
toml  0.5.8    ---     0.5.9   Normal  ---
```
</details>

@fpoli could you take care of this?

### CODE REVIEW

**Comment by fpoli** [comment]
Status: Prusti reports internal errors like "Non-slice LHS type '&[i32]' not supported" because `.is_slice()` on `&[i32]` seems to return `false`.

**Comment by Aurel300** [comment]
We can use [`is_array_slice`](https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_middle/ty/sty.rs.html#1899) although this also considers some raw pointers as slices. Alternatively we could `.peel_refs().is_slice()` (if we don't already)?

File: prusti-viper/src/encoder/mir/pure/pure_functions/interpreter.rs [hunk 1]
```diff
@@ -1172,10 +1172,10 @@ impl<'p, 'v: 'p, 'tcx: 'v> BackwardMirInterpreter<'tcx>
...
 
                     mir::Rvalue::Cast(mir::CastKind::Pointer(ty::adjustment::PointerCast::Unsize), ref operand, lhs_ref_ty) => {
-                        let lhs_ty = lhs_ref_ty.peel_refs();
                         let rhs_ref_ty = self.mir_encoder.get_operand_ty(operand);
+                        let lhs_ty = lhs_ref_ty.peel_refs();
                         let rhs_ty = rhs_ref_ty.peel_refs();
-                        if lhs_ref_ty.is_slice() && rhs_ty.is_array() {
+                        if lhs_ty.is_slice() && rhs_ty.is_array() {
                             let function_name = self.encoder.encode_unsize_function_use(rhs_ty, lhs_ty)
                                 .unwrap_or_else(|error| unreachable!("error during unsizing slice to array: {:?}", error) );
```
Comment (Pointerbender) [inline prusti-viper/src/encoder/mir/pure/pure_functions/interpreter.rs]: Isn't this message supposed to be the other way around? (`"error during unsizing array to slice: {:?}"`)

Comment (Pointerbender) [inline prusti-viper/src/encoder/mir/pure/pure_functions/interpreter.rs]:
```suggestion
                                .unwrap_or_else(|error| unreachable!("error during unsizing array to slice: {:?}", error) );
```

Comment (fpoli) [reply prusti-viper/src/encoder/mir/pure/pure_functions/interpreter.rs]: It seems so, thanks! I'm about to force-push, so I'll manually include your suggestion.

**Comment by fpoli** [comment]
Mmm, there are some "Prusti: unsupported feature" errors that have been introduced by recent changes to `master`.

**Pull request closed without merging**